### DISCUSSION TITLE
Deleting lines of code which should not be deleted.

### DISCUSSION THREAD

**Discussion by gudata** [astral-sh/ruff#9947] [Q&A] [answered]
I was in the middle of refactoring. When I save the file, it's ok. but when I add blank line before the logger_link_scrape and save, the```logger_link_scrape = logging.getLogger("link-scrape")``` is deleted.

I need that line :) why that happens?

```
# pylint: disable=C0116,C0115
import logging
import time
from datetime import datetime
from pprint import pprint
from queue import PriorityQueue
from urllib.parse import urldefrag, urlparse

logger_link_scrape = logging.getLogger("link-scrape")
```

❯ ruff check crawler.py
app/website/indexer/crawler.py:2:8: F401 [*] `logging` imported but unused

**Comment by MichaReiser** [comment]
Can you tell us a bit more about your set up? I assume you're using ruff in VS code. Do you have `source.fixAll` enabled by chance?

```json
{
  "[python]": {
    "editor.codeActionsOnSave": {
      "source.fixAll": true
    }
  }
}
```

**Reply by gudata** [reply]
It is vscode. I didn't have the  "source.fixAll": true.
 
 It was the isort extension.
 
 Thank you for helping!

**Comment by charliermarsh** [comment, marked as answer]
My guess is that you're running into https://github.com/astral-sh/ruff-vscode/issues/128 which is a VS Code bug (https://github.com/microsoft/vscode/issues/174295). Can you try removing the isort extension?

**Reply by gudata** [reply]
After uninstall the isort everything works again! 

Thank you!

**Reply by KrzysztofPrzygodzki** [reply]
Same issue, same resolution. Maybe add a warring - "Conflict with isort!"?
